bip-0119.2.mediawiki

```diff
@@ -0,0 +1,184 @@
+<pre>
+  BIP: 119.2 (Pending Assignment)
+  Layer: Consensus (soft fork)
+  Title: CHECKTEMPLATEVERIFY V2
+  Author: Jeremy Rubin <j@rubin.io>
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-0119
+  Status: Draft
+  Type: Standards Track
+  Created: 2024-05-06
+  License: BSD-3-Clause
+</pre>
+
+==Abstract==
+
+This BIP proposes extensions to OP_CHECKTEMPLATEVERIFY BIP-119 as enhancements
+for LN Symmetry based constructions which use CTV and CSFS.
+
+
+==Summary==
+
+Adds a 20-byte CTV hash using HASH160 digest to save 12 bytes per script.  Adds
+a 1-byte flag (with no bits set) that allows for arbitrary data to be committed
+to inside the CTV hash.
+
+==Motivation==
+
+LN Symmetry prefers scripts as small as possible. Allowing the use of HASH160
+can allow users to reduce fees in exchange for minor loss of security.
+
+LN Symmetry also sometimes requires additional committed data for data
```

Comment:
One of the things I'm noticing here is that the extra data is always a byte. I'm not 100% clear on how LN-Sym uses this as I forget the details but it occurs to me that you need not limit yourself to a single byte as you can just treat any data in excess of the maximum interpretable hash size as extra data without ambiguity:

20 bytes = Hash160 + 0 bytes
25 bytes = Hash160 + 5 bytes
33 bytes = Hash256 + 1 byte
100 bytes = Hash256 + 68 bytes

Whether this flexibility is useful or not I'm unsure as I would need to re-read the LN-Sym literature in detail.